DataDog/datadog-ci-rb
# [SDTEST-116] Send event_created and event_finished metrics for internal telemetry

**What does this PR do?**
Adds `event_created` and `event_finished` metrics. As this PR is already large, I will implement other metrics in subsequent PRs.

Also this PR:
- only enables telemetry with Datadog Agent for now as agentless mode is not released yet in gem datadog
- refactors domain model code
- introduces telemetry helpers and constants
- bumps datadog dependency to 2.2 as this is currently the oldest version with metrics support
- adds minor changes to dev setup: run bundle in docker without bundler version, replace `docker-compose` command with `docker compose` command

**How to test the change?**
Tested by running test suite with staging settings, see (DD only) dashboard [here](https://dd.datad0g.com/dashboard/xgg-hpe-mre/ci-visibility-telemetry-metrics?fromUser=true&refresh_mode=sliding&view=spans&from_ts=1721133982924&to_ts=1721738782924&live=true).

<img width="1325" alt="image" src="https://github.com/user-attachments/assets/2eb62bf0-ecf6-489f-9e71-a42f8cb2e0b4">
